### DISCUSSION TITLE
EJBCA 8.2.0.1 recommends to use Wildfly 26.1.3.Final but it's a very old version

### DISCUSSION THREAD

**Discussion by fco-code-org** [Keyfactor/ejbca-ce#574] [General]
Hello,

The recommended version which is 26.1.3 has a lot high CVEs :-(
The more recent Wildfly version is 31.0.1.Final.

Is there some plan to upgrade the recomended Wildfly version for EJBCA  ?

Maybe the EJBCA Enterprise version recommends more recent Wildfly version ?  

Best Regards.

**Comment by primetomas** [comment]
See arlier discussions #569

**Reply by fco-code-org** [reply]
Hello,
I performed a quick search about Jakarta compliancy.
So:

- WildFly 18.0.0 is Jakarta 8  certified
- Wildfly 23.0.2 and 25.0.0 are Jakarta 9.1 certified
- Wildfly 27.0.0 is Jakarta 10 certified.

The question then is: what is the EJBCA  Jakarta certification level ?
As the recommended version is 26.1.3 then EJBCA 8.2.0.1 should be compliant with at least Jakarta 9.1, right ?

Now, the question is : when EJBCA will be compliant with Jakarta 10 in order to be recommended on more recent Wildfly version, right ?

Best Regards.

**Reply by primetomas** [reply]
It says in #569

**Reply by primetomas** [reply]
It also mentions that JBoss is not Jakarta EE 10 yet, they will hopefully come out with a new release in the fall as well.

**Reply by fco-code-org** [reply]
Hello,

Something I don't understand.
EJBCA could use JBoss or Wildfly, right ?

Wildfly 27.0.0.Alpha5 is Jakarta 10 certified, so I guess that the 27.0.0.Final version also is, right ?
So, the recomended Wildfly version could be then Wildfly 27.0.0.Final, right ?

Best Regards.

**Reply by primetomas** [reply]
No, EJBCA is Jakarta 9, in order to be able to run both on WIldFly and JBoss. Being able to run on WildFly 31 requires breaking changes so the same version of EJBCA could not run on JBoss.
We don't want to maintain two completely separate softwares, which is why we are holding off until all users can move to an application server that supports Jakarta 10.

**Reply by mike-agrenius-kushner** [reply]
Ahum, EJBCA is at Jakarta8, ya noob :P 

Nobody bothered with Jakarta9.

JBoss EAP 8 has been out since February, and I believe it's based on WF30, so it's Jakarta10 compatible. 

Problem was that Redhat were far from forthcoming with when the release was actually coming, so as Tomas said we either had to branch EJBCA into two versions or stick to WF26 in order to not alienate JBoss users. Now that it's out (and we had time to finish off what we'd started), a conversion to Jakarta10 is in the works. Breaking backwards compatibility in Jakarta10 kind of sucked :/

**Reply by fco-code-org** [reply]
Hello Guys,

Sorry I don't understand you.
As you said JBoss 8 based on Wildfly 30 so it is Jakarta 10 compatible. 

So we have:
    -  Wildfly 30 (base of JBoss 8) which is Jakarta 10 compatible
    - JBoss 8 which is Jakarta 10 compatible  

Maybe I over simplified but we might have a single EJBCA version compatible with either JBoss8 and Wildfly 30. 

Best regards.

**Reply by mike-agrenius-kushner** [reply]
Yeah, exactly. It's been out for under two months and Redhat gave no prior warning to the release. Moving from Jakarta8 to Jakarta10 is a major refactoring of our code, so it's going to be a few months until we put out EJBCA 9, which is going to be Jakarta10.

**Reply by fco-code-org** [reply]
Hello,

As far as I have understood the gap is less important I would say as EJBCA (8.2.0.1) is currently Jakarta 9 or 9.1  compatible if I am not mistaken.
So the gap to fill is between Jakarta 9 or 9.1 and Jakarta 10 ;-) which would be less huge that filling the gap between Jakarta 8 and Jakarta 10, I guess.

One suggestion: name it EJBCA 10. Then it will be understandable for everyone that this EJBCA version is in line with this Jakarta version. 

Best regards.

**Reply by mike-agrenius-kushner** [reply]
No, EJBCA is at JEE8/Jakarta8.

As you can see from https://docs.wildfly.org/, Wildfly essentially skipped proper support for Jakarta9 and skipped straight to 10 with WF27. 

And believe me, I did the conversion as a PoC, it's a ton of work and nothing we could plan in on our roadmap in a moment's notice.

**Reply by fco-code-org** [reply]
Hello,

It's misleading. 
Looking at https://jakarta.ee/compatibility/certification/9.1/:  Wildfly preview 23.0.2 and 25.0.0 are Jakarta 9.1 compatible.
Implicitly, my guess is that Wildfly 26.1.3 might be Jakarta 9.1 compatible. 

But according to the link you provided my guess is wrong: Wildfly 26.1.3 is just Jakarta 9.1 preview :-( 

Best Regards.
